elastic/beats
# [MetricBeat] add param `aws_partition` to support aws-cn, aws-us-gov regions

## What does this PR do?

This PR is to add a parameter `aws_partition` for metricbeat aws cloudwatch module, so metricbeat can support aws china and aws us-gov account.



## Why is it important?

Because there are three partitions ` aws, aws-cn, aws-us-gov` of aws regions,  now metricbeat can only support aws global region, here is a hardcode. 


https://github.com/elastic/beats/blob/5ba7026f81afcdf9a2d745eb248d2711d901673e/x-pack/metricbeat/module/aws/aws.go#L95-L100



## Checklist





## How to test this PR locally



1. add `aws_partition` param in aws module config 

```
metricbeat.modules:
  - module: aws
    aws_partition: aws-cn
    period: 300s
    access_key_id: 
    secret_access_key:
```
2. run metricbeat , open debug log, then you can see it can retrive metrics from correct regions



## Related issues



-  Closes elastic/beats#18850

## Use cases



* no side affect
* user can use metricbeat to get metrics from aws china and aws us-gov account except aws global account


## Screenshots



aws global account, set `aws_partition: aws ` in aws module config  or remove the param

![image](https://user-images.githubusercontent.com/12824991/85829385-13f8dd00-b7bd-11ea-8bbe-002f4de59325.png)

 aws china account, set `aws_partition: aws-cn ` in aws module config 

![image](https://user-images.githubusercontent.com/12824991/85829463-40acf480-b7bd-11ea-9ab5-536e79992325.png)
